lib/xlat_tables/xlat_tables_common.c

```diff
@@ -284,8 +284,9 @@ static mmap_region_t *init_xlation_table_inner(mmap_region_t *mm,
 	unsigned level_size_shift = L1_XLAT_ADDRESS_SHIFT - (level - 1) *
 						XLAT_TABLE_ENTRIES_SHIFT;
 	unsigned level_size = 1 << level_size_shift;
-	unsigned long long level_index_mask = XLAT_TABLE_ENTRIES_MASK <<
-							level_size_shift;
+	unsigned long long level_index_mask =
+		((unsigned long long) XLAT_TABLE_ENTRIES_MASK)
+		<< level_size_shift;
```

Comment:
(At the moment, init_xlation_table_inner() is the only place where 
XLAT_TABLE_ENTRIES_MASK is used so there is no additional user for now.)

I agree suffixing the constant with ULL would work as well but it is not necessarily what
one wants in all cases. XLAT_TABLE_ENTRIES_MASK equals 0x1ff, which easily
fits in 32 bits. Whether we need to cast it to a 64-bit value depends on the context in
which it is used so I think the cast should remain wherever the constant is used, not
in the definition itself.